src/FSAL/FSAL_CORTXFS/mds.c

```diff
@@ -269,25 +280,147 @@ int kvsfs_pmds_ini(struct kvsfs_fsal_module *kvsfs,
 		goto out;
 	}
 
-	mds_ctx->pnfs_role = KVSFS_PNFS_DISABLED;
-	mds_ctx->mds = gsh_calloc(1, sizeof(struct kvsfs_pnfs_mds_info));
-	if (mds_ctx->mds == NULL) {
-		LogCrit(COMPONENT_PNFS,
+
+	/* Read the pNFS.conf file */
+	rc = config_from_file("libcortxfs", CONF_FILE, &cfg_items,INI_STOP_ON_ERROR, &errors);
+	if(rc) {
+			free_ini_config_errors(errors);
+			rc = -rc;
+	} 
+
+	/* Read the configuration parameters */
+	rc = get_config_item("pNFS","pNFS_role",cfg_items, &items);
+	if (items==NULL) {
+		LogCrit(COMPONENT_PNFS,"pNFS parameters configured incorrectly\n");
+		rc = -rc;
+		goto out; 
+	}
+	pNFS_role = get_string_config_value(items, NULL);
+	if (strncmp(pNFS_role,"MDS",sizeof("MDS")) == 0) {
+		mds_ctx->pnfs_role = CORTXFS_PNFS_MDS;
+	} 
+	else if (strncmp(pNFS_role,"DS",sizeof("DS")) == 0) { 
+		mds_ctx->pnfs_role = CORTXFS_PNFS_DS;
+	}
+	else if (strncmp(pNFS_role,"BOTH",sizeof("BOTH")) == 0) {
+		mds_ctx->pnfs_role = CORTXFS_PNFS_BOTH;
+	}
+	else {
+		mds_ctx->pnfs_role = CORTXFS_PNFS_DISABLED;
+	}
+	
+	
+	if (mds_ctx->pnfs_role > CORTXFS_PNFS_DS )
+	{	
+		mds_ctx->mds = gsh_calloc(1, sizeof(struct cortxfs_pnfs_mds_info));
+		if (mds_ctx->mds == NULL) {
+			LogCrit(COMPONENT_PNFS,
 			"gsh_calloc failed: %u", sizeof(*mds_ctx->mds));
-		rc = -ENOMEM;
-		goto out;
+			rc = -ENOMEM;
+			goto out;
+		}
+		mds_ctx->mds->stripe_unit = CORTXFS_PNFS_STRIPE_UNIT;	
+	
+		rc = get_config_item("pNFS","NumDataServers",cfg_items, &items);
+		if (items == NULL){
+			LogCrit(COMPONENT_PNFS,"Zero data servers\n");
+			rc = -rc; 
+			goto out;
+		}
+		
+		num_ds=(int)get_int_config_value(items, 0, 0, NULL);
+		
+		LIST_INIT(&mds_ctx->mds->mds_ds_list);
+			
+		char DS_List[256],tmp[256];
+                char *DS_Info,*port,*ipaddr;
+		int i;	
```

Comment:
Please put the declarations in the beginning of this function.

Reply:
done